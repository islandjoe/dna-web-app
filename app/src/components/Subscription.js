import React, {Component} from 'react'
import styled from 'styled-components'
import {packages, subscribers} from '../data'

class Subscription extends Component {

  constructor(props) {
    super(props)

    this.state = {
      package: {
        id:    '',
        name:  '',
        type:  '',
        price: 0
      },
      subscriber: {
        userid: ''
      }
    }
  }

  componentDidMount() {
    const packageId = this.props.match.params.id
    const subscription = packages.find((p)=>
      p.id === packageId)

    if (subscription) {
      this.setState({
        package: { ...subscription }
      })
      this.setState({ package: {...subscription} })
    }

    const subscriber = subscribers.find((s)=> s.userid ===  this.props.userid)

    if (subscriber) {
      this.setState({
        subscriber: { ...subscriber }
      })
      this.setState({ subscriber: {...subscriber} })
    }

  } //componentDidMount

   render() {
    const {
      package: bundle,
      subscriber: user
    } = this.state

    return (
      <>
      <header>
        { bundle.name }
      </header>

      <table>
        <Details>
        <tr>
          <th>User Id</th>
          <td className='UserId'>
            { user.userid }
          </td>
        </tr>
        {// Hard-code this comparison for now:
          bundle.type == 'Mobile'
        ?
          <tr>
            <th>Mobile</th>
            <td className='Mobile'>{ user.mobile }</td>
          </tr>
          :
          <tr>
            <th>Address</th>
            <td className='Address'>{ user.address }</td>
          </tr>
        }
        <tr>
          <th>Subscription Type</th>
          <td className='SubscriptionType'>
            { bundle.type }
          </td>
        </tr>
        <tr>
          <th>Monthly price</th>
          <td className='MonthlyPrice'>
            { `${ bundle.price }â‚¬` }
          </td>
        </tr>
        </Details>
      </table>
      </>
    )
  } //render
} //Subscription

const Details = styled.tbody` `

export default Subscription
