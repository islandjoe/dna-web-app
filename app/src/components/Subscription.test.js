import React from 'react'
import {shallow} from  'enzyme'
import Subscription from './Subscription.js'

it('should render correctly', ()=> {
  const component = shallow(<Subscription/>)
  expect(component).toMatchSnapshot()
})