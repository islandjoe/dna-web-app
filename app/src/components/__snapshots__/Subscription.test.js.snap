// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<Fragment>
  <h3>
    Home Broadband
  </h3>
  <table>
    <styled.tbody>
      <tr>
        <styled.th>
          User name
        </styled.th>
        <styled.td>
          i.am@island.joe
        </styled.td>
      </tr>
      <tr>
        <styled.th>
          Address
        </styled.th>
        <styled.td>
          727 Bristol Street, Trexlertown, Arizona, 7038
        </styled.td>
      </tr>
      <tr>
        <styled.th>
          Subscription type
        </styled.th>
        <styled.td>
          1 year
        </styled.td>
      </tr>
      <tr>
        <styled.th>
          Monthly price
        </styled.th>
        <styled.td>
          26â‚¬
        </styled.td>
      </tr>
    </styled.tbody>
  </table>
</Fragment>
`;
